cmd/utils.go

```diff
@@ -349,3 +350,70 @@ func parseAttribute(attrs string) (map[string]string, error) {
 
 	return attribute, nil
 }
+
+// EqualEvents tells whether a and b contain the same events
+func EqualEvents(a, b []minio.NotificationEventType) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	sort.SliceStable(a, func(i, j int) bool {
+		return a[i] < a[j]
+	})
+	sort.SliceStable(b, func(i, j int) bool {
+		return b[i] < b[j]
+	})
+	for i, v := range a {
+		if v != b[i] {
+			return false
+		}
+	}
+	return true
+}
+
+// EqualFilterRules tells whether a and b contain the same filters
+func EqualFilterRules(a, b []minio.FilterRule) bool {
```

Comment:
You should reimplement this to use set.NewStringSet() instead such that you can use them to ensure 

```
a.Intersection(b)
```

this way you can generalize the overall approach